File: frontend/src/api/coverLetter.ts
import axios from "axios";

const API_BASE = import.meta.env.VITE_API_URL || "http://localhost:8000";

export interface CoverLetterRequest {
  resume_id: number;
  job_id: number;
  tone: "formal" | "casual" | "creative";
}

export interface CoverLetterResponse {
  cover_letter: string;
  tone: string;
  company: string;
  role: string;
}

export async function generateCoverLetter(
  payload: CoverLetterRequest,
  token: string
): Promise<CoverLetterResponse> {
  const { data } = await axios.post<CoverLetterResponse>(
    `${API_BASE}/cover-letter/generate`,
    payload,
    { headers: { Authorization: `Bearer ${token}` } }
  );
  return data;
}

File: backend/app/schemas/cover_letter.py
from pydantic import BaseModel
from typing import Literal


class CoverLetterRequest(BaseModel):
    resume_id: int
    job_id: int
    tone: Literal["formal", "casual", "creative"] = "formal"


class CoverLetterResponse(BaseModel):
    cover_letter: str
    tone: str
    company: str
    role: str